File: src/main/kotlin/interview_150/two_pointer/IsSubsequence.java
package interview_150.two_pointer;

/**
 * Difficulty : Easy
 * Problem Statement : https://leetcode.com/problems/is-subsequence/description/?envType=study-plan-v2&envId=top-interview-150
 */

public class IsSubsequence {
    public boolean isSubsequence(String s, String t) {
        int sp = 0;
        int tp = 0;

        while (sp < s.length() && tp < t.length()) {
            if (s.charAt(sp) == t.charAt(tp)) {
                sp++;
            }
            tp++;
        }
        return sp == s.length();
    }
}

File: src/main/kotlin/leetcode_75/level_1/day_2/IsSubsequence.kt
package leetcode_75.level_1.day_2


/**
 * Difficulty : Easy
 * Problem Statement : https://leetcode.com/problems/is-subsequence/
 */

fun main() {
    println(IsSubsequence().isSubsequence("abc", "ahbgdc"))
}

class IsSubsequence {
    fun isSubsequence(s: String, t: String): Boolean {
        var i = 0
        var j = 0

        while (i < s.length && j < t.length) {
            if (s[i] == t[j]) i++
            j++
        }

        return i == s.length
    }
}